easybuild/easyblocks/o/openmpi.py

```diff
@@ -167,25 +168,28 @@ def load_module(self, *args, **kwargs):
     def sanity_check_step(self):
         """Custom sanity check for OpenMPI."""
 
+        version = LooseVersion(self.version)
+        use_internal_prrte = version >= '5.0.0' and 'PRRTE' not in self.cfg.dependency_names()
+
         bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'mpirun', 'ompi_info', 'opal_wrapper']
-        if LooseVersion(self.version) >= LooseVersion('5.0.0'):
-            if not get_software_root('PRRTE'):
+        if version >= '5.0.0':
+            if use_internal_prrte:
```

Comment:
can be simplified to just `if use_internal_prrte`?

Reply:
No, the 5.0 check has an else, same below.
I included the 5.0 check in internal_prrte even though it can only be used once, so that at the point of definition it is clear that this is 5.0-only

Reply:
@Flamefire thanks, I realise now that I could have been a little more specific :)